examples/src/test/kotlin/AggregatesBuilderTest.kt

```diff
@@ -0,0 +1,939 @@
+                    UnwindOptions().preserveNullAndEmptyArrays(true)
+                )
+                // :snippet-end:
+            ))
+        assertEquals(10, resultsFlow.toList().size)
+    }
+
+    @Test
+    fun unwindOptionsArrayTest() = runBlocking {
+        val resultsFlow = movieCollection.aggregate<Document>(listOf(
+            // aggregate content to unwind
+            Aggregates.group("\$${Movie::year.name}", Accumulators.bottom("lowestRatedTwoMovies", Sorts.descending("${Movie::imdb.name}.${Movie.IMDB::rating.name}"), listOf("\$${Movie::title.name}", "\$${Movie::imdb.name}.${Movie.IMDB::rating.name}"))),
+            // :snippet-start: unwind-options-array
+                Aggregates.unwind(
+                    "\$${"lowestRatedTwoMovies"}",
+                    UnwindOptions().includeArrayIndex("position")
+                )
+                // :snippet-end:
+            ))
+        assertEquals(10, resultsFlow.toList().size)
+    }
+
+    @Test
+    fun outTest() = runBlocking {
+        val resultsFlow = movieCollection.aggregate<Document>(listOf(
+            // aggregate content to push out
+            Aggregates.group("\$${Movie::year.name}", Accumulators.bottom("lowestRatedTwoMovies", Sorts.descending("${Movie::imdb.name}.${Movie.IMDB::rating.name}"), listOf("\$${Movie::title.name}", "\$${Movie::imdb.name}.${Movie.IMDB::rating.name}"))),
+                // :snippet-start: out
+                Aggregates.out("classic_movies"),
+                // :snippet-end:
+            ))
+        assertEquals(5, resultsFlow.toList().size)
+    }
+
+    @Test
+    fun mergeTest() = runBlocking {
+        val resultsFlow = movieCollection.aggregate<Document>(listOf(Aggregates.match(Filters.and(Filters.gte(Movie::year.name, 1990), Filters.lt(Movie::year.name, 2000))),
+                // :snippet-start: merge
+                Aggregates.merge("nineties_movies")
+                // :snippet-end:
+            ))
+        val results = resultsFlow.toList()
+        assertEquals(6, resultsFlow.toList().size)
+        assertEquals(6, results.filter { it.getInteger(Movie::year.name) in (1990..1999) }.size)
+    }
+
+    @Test
+    fun mergeOptionsTest() = runBlocking {
+        val uniqueIndexOption = IndexOptions().unique(true)
+        val movieRatings = database.getCollection<Document>("movie_ratings")
+        movieRatings.createIndex(Indexes.ascending("year", "title"), uniqueIndexOption)
+
+        val resultsFlow = movieCollection.aggregate<Document>(listOf(
+            Aggregates.match(Filters.eq(Movie::title.name, "The Sixth Sense")),
+            Aggregates.project(Projections.fields(Projections.computed("rating", "\$imdb.rating"), Projections.include("title", "year"))),
+            // :snippet-start: merge-options
+            Aggregates.merge(
+                MongoNamespace("aggregation", "movie_ratings"),
+                MergeOptions().uniqueIdentifier(listOf("year", "title"))
```

Comment:
should this be references to property names, like:

```suggestion
                MergeOptions().uniqueIdentifier(listOf(Movie::year.name, Movie::title.name))
```